### PR TITLE
Feature/gfi multiple features

### PR DESCRIPTION
**Pull request by dopenguin**
## Summary

Add the configuration parameters `directSelect` and `multiSelect` to be able to select multiple features.

## Instructions for local reproduction and review

`npm run snowbox` and use a modifier key to drag and select multiple. If you drag across a selected value, it gets deselected. If you configure `directSelect`, it should work the same. Beware the information in the readme though.

### CODE REVIEW

**Review by warm-coolguy — CHANGES_REQUESTED** [review]
🏓 @dopenguin

File: packages/plugins/Gfi/src/store/actions/debouncedGfiRequest.ts [hunk 1]
```diff
@@ -82,27 +86,47 @@ const getPromisedFeatures = (
...
+        Array.isArray(features) && Array.isArray(oldSelection[layerId])
+          ? features.reduce((acc, newFeature) => {
+              // If the feature is already in the old selection, remove it
+              const oldFeatureIndex = acc.findIndex((oldFeature) =>
+                compare(oldFeature.properties, newFeature.properties)
+              )
+              return oldFeatureIndex === -1
+                ? [...acc, newFeature]
+                : acc.filter((_, i) => i !== oldFeatureIndex)
+            }, oldSelection[layerId] as GeoJsonFeature[])
```
[the remark this answers is not in the document]
Comment (dopenguin) [reply packages/plugins/Gfi/src/store/actions/debouncedGfiRequest.ts:124]: My guess would've been that 

```ts
Array.from(
  new Set(features).symmetricDifference(
    new Set(oldSelection[layerId])
  )
)
```

would've worked, but it just leads to the situation that everything gets added, constantly.  
My guess would be that `symmetricDifference` can not compare the features properly.

File: packages/plugins/Gfi/src/store/actions/index.ts [hunk 2]
```diff
@@ -78,6 +80,29 @@ export const makeActions = () => {
         )
       }
     },
+    setupMultiSelection({ dispatch, getters, rootGetters }) {
```
[the remark this answers is not in the document]
Comment (dopenguin) [reply packages/plugins/Gfi/src/store/actions/index.ts:83]: Agreed and adjusted da374f2a68c1e0f85757e03ff1788b0346ab6c34

**Comment by dopenguin** [comment]
> 🏓 @dopenguin

🏓 @warm-coolguy

File: packages/plugins/Gfi/src/store/actions/debouncedGfiRequest.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (warm-coolguy) [reply packages/plugins/Gfi/src/store/actions/debouncedGfiRequest.ts:124]: Revisiting this, I've realized the comparison only works by the features' properties. So we're probably looking at copies, which would explain why `symmetricDifference` does not work.

I wonder whether that's a problem for a general solution. On the one hand, our use cases will probably always be distinguishable by their properties. On the other, we may have ids or even [foreign members](https://datatracker.ietf.org/doc/html/rfc7946#section-6.1) (had to look this up myself ...) as distinguishing elements.

Hence I'd propose we at least document that feature ought to be distinguishable by their properties for the functionality to work, and, if it ever becomes a problems, revisit this.

**Comment by warm-coolguy** [comment]
> > 🏓 @dopenguin
> 
> 🏓 @warm-coolguy

🏓 @dopenguin

File: packages/plugins/Gfi/src/store/actions/debouncedGfiRequest.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (dopenguin) [reply packages/plugins/Gfi/src/store/actions/debouncedGfiRequest.ts:124]: Added a relevant comment to the documentation d4a0d109c4ee5f814a3a8911df1d28a0451b6e5f

**Pull request merged**